data/World/Shadow Temple.json

```diff
@@ -116,10 +138,28 @@
         "region_name": "Shadow Temple Before Boss",
         "dungeon": "Shadow Temple",
         "locations": {
+            "Shadow Temple After Boat Upper Recovery Heart 1" : "can_use(Distant_Scarecrow)",
+            "Shadow Temple After Boat Upper Recovery Heart 2" : "can_use(Distant_Scarecrow) ",
+            "Shadow Temple After Boat Lower Recovery Heart" : "can_use(Distant_Scarecrow) or ((Bow or (logic_shadow_statue and has_bombchus) or shadow_temple_shortcuts) and can_play(Song_of_Time))",
             "Shadow Temple Spike Walls Left Chest": "can_use(Dins_Fire)",
             "Shadow Temple Boss Key Chest": "can_use(Dins_Fire)",
+            "Shadow Temple Spike Walls Pot": "True",
             "Shadow Temple Invisible Floormaster Chest": "True",
-            "Shadow Temple GS Triple Giant Pot": "True"
+            "Shadow Temple Invisible Floormaster Pot 1" : "True",
+            "Shadow Temple Invisible Floormaster Pot 2" : "True",
+            "Shadow Temple After Boat Pot" : "True",
+            "Shadow Temple Near Boss Pot 1" : "(Bow or can_use(Distant_Scarecrow) or (logic_shadow_statue and has_bombchus) or shadow_temple_shortcuts)",
+            "Shadow Temple Near Boss Pot 2" : "(Bow or can_use(Distant_Scarecrow) or (logic_shadow_statue and has_bombchus) or shadow_temple_shortcuts)",
+            "Shadow Temple GS Triple Giant Pot": "True",
+            "Shadow Temple 3 Spinning Pots Rupee 1": "Bomb_Bag or Progressive_Strength_Upgrade",
+            "Shadow Temple 3 Spinning Pots Rupee 2": "Bomb_Bag or Progressive_Strength_Upgrade",
+            "Shadow Temple 3 Spinning Pots Rupee 3": "Bomb_Bag or Progressive_Strength_Upgrade",
+            "Shadow Temple 3 Spinning Pots Rupee 4": "Bomb_Bag or Progressive_Strength_Upgrade",
+            "Shadow Temple 3 Spinning Pots Rupee 5": "Bomb_Bag or Progressive_Strength_Upgrade",
+            "Shadow Temple 3 Spinning Pots Rupee 6": "Bomb_Bag or Progressive_Strength_Upgrade",
+            "Shadow Temple 3 Spinning Pots Rupee 7": "Bomb_Bag or Progressive_Strength_Upgrade",
+            "Shadow Temple 3 Spinning Pots Rupee 8": "Bomb_Bag or Progressive_Strength_Upgrade",
+            "Shadow Temple 3 Spinning Pots Rupee 9": "Bomb_Bag or Progressive_Strength_Upgrade"
```

Comment:
This can be done with Bombchus. We have a trick for the single pot earlier in the temple, but obviously it's a lot harder when they're moving.